### PR TITLE
Add 2.0 release logic

### PR DESCRIPTION
**Pull request by bmw**
This PR:

* Deletes the 2.0 pre-release pipeline
* Causes 1.x releases to be released to Docker Hub without updating the latest tag, PyPI, and the candidate and stable channels of the snap store
* Causes 2.x releases to be released to Docker Hub, PyPI, the beta channel of the snap store, and our Windows installer

We could potentially look into how to continue to do 1.x Windows installer releases through GitHub releases and tech ops tooling, but I personally don't think it's worth it right now.

This PR **DOES NOT** do anything about progressive snap releases. I think we can revisit this when/if we decide (how) to do them.

Testing done:

* A [dummy 1.0 release](https://dev.azure.com/certbot/certbot/_build/results?buildId=6078&view=results)
* A [dummy 2.0 release](https://dev.azure.com/certbot/certbot/_build/results?buildId=6076&view=results)
* A dummy 3.0 release which immediately errors
* Running this modified version of `finish_release.py` identifies that the last release was for 1.32.0 and redeploys the snaps.

After this PR is merged, I think we should:

- [ ] Delete the [dummy release pipeline](https://dev.azure.com/certbot/certbot/_build?definitionId=8) I created
- [ ] Delete the 2.0 pre-release instructions, [the associated pipeline](https://dev.azure.com/certbot/certbot/_build?definitionId=7), the `2.0.x` git branch (since it'll be recreated as part of the release), and any special GitHub protections for the `2.0.x` branch in favor of our normal settings for point release branches.
- [ ] Update the release instructions to say that `finish_release.py` will only prompt for CSS related steps if you're not doing a 1.x release
- [ ] Add text somewhere explaining our current 2.x release strategy especially with regards to snaps. I recommend just adding something to the bottom of the changelog for 2.0.0 so it can be seen there and in our release announcements.
- [ ] Cherry pick the commit created by this PR to the 1.32.x branch
- [ ] Do our first stable 2.0 release!

### CODE REVIEW

File: tools/finish_release.py [hunk 1]
```diff
@@ -195,9 +186,9 @@ def main(args):
     # again fails. Publishing the snaps can be done multiple times though
     # so we do that first to make it easier to run the script again later
     # if something goes wrong.        
-    if parsed_args.publish_snaps:
+    if version.startswith('1.'):
```
Comment (alexzorin) [inline tools/finish_release.py:189]: Maybe an explanatory comment here to explain the strategy and effect? Without context, the code isn't super clear about what it's doing.

**Review by alexzorin — CHANGES_REQUESTED** [review]
Other than one comment, this LGTM. Another review from Will or Erica is probably a good idea.

**Comment by wgreenberg** [comment]
+1 from me!

**Review by alexzorin — APPROVED** [review]

**Comment by bmw** [comment]
Thanks for the quick reviews!

**Pull request merged**